### ISSUE TITLE
how to remove or show an uploaded *invisible* theme?

### ISSUE THREAD

**Issue by tkimnguyen** [plone/plone.app.theming#107]
related to #105, I have a site in which I uploaded a barebones theme but because it is missing manifest.cfg I don't see it in the top-level @@theming-controlpanel and cannot therefore remove it.

**Comment by davilima6** [comment]
I think this a duplicate from plone/Products.CMFPlone#1622.

**Comment by tkimnguyen** [comment]
Yes, @3dogMcNeill and I were trying to figure this out this morning. She filed the software issue and I filed the documentation issue, but now I'm trying to be thorough and cover all the other related problems.

**Comment by tkimnguyen** [comment]
It's only possible to recover from the bug by remembering which URL to browse to. There should be a way for @@theming-controlpanel to show all themes, even if they don't have a manifest.cfg. Or, I guess, fix the bug so it's not possible to upload a theme without manifest.cfg or to auto generate a manifest.cfg, but that still leaves some people in the position of having an invisible theme that prevents them cryptically/silently from uploading a new theme with the same name.

**Comment by davilima6** [comment]
The theme is only invisible because manifest.cfg isn't there, so if it gets autogenerated we won't have the silent fail anymore.

**Comment by davilima6** [comment]
Btw another way to recover is going to /portal_resources and deleting the "theme/YOURTHEME" folder.

**Comment by tkimnguyen** [comment]
Probably the upload mechanism needs to create a default manifest.cfg if one is not included in the uploaded zip.

**Comment by tkimnguyen** [comment]
Could replicate this code that creates a `rules.xml` file if it is missing from the theme, and do the same for `manifest.cfg` which is also already in the template theme.

**Comment by VerSooSet** [comment]
There is a template generation of manifest file. It's a my part to GSoC'17. This solves the problem?
https://github.com/VerSooSet/plone.app.theming/tree/as-rules-manifest

**Comment by tkimnguyen** [comment]
I started a PR to do the checking for a manifest.cfg and to add a default one if it is missing. Not sure when I can finish it.
